go.mod

```diff
@@ -17,17 +22,25 @@ require (
 	github.com/spf13/pflag v1.0.5
 	github.com/spf13/viper v1.6.1
 	github.com/stretchr/testify v1.4.0
+	golang.org/x/net v0.0.0-20200301022130-244492dfa37a // indirect
+	golang.org/x/sys v0.0.0-20200302150141-5c8b2ff67527 // indirect
+	golang.org/x/xerrors v0.0.0-20191204190536-9bdfabe68543 // indirect
 	gopkg.in/h2non/gock.v1 v1.0.14
-	gopkg.in/yaml.v2 v2.2.4
+	gopkg.in/yaml.v2 v2.2.8
 	k8s.io/api v0.17.3
 	k8s.io/apiextensions-apiserver v0.17.3
 	k8s.io/apimachinery v0.17.3
 	k8s.io/client-go v12.0.0+incompatible
 	k8s.io/klog v1.0.0
+	k8s.io/kube-openapi v0.0.0-20200204173128-addea2498afe // indirect
+	k8s.io/utils v0.0.0-20200229041039-0a110f9eb7ab // indirect
 	sigs.k8s.io/controller-runtime v0.5.0
 	sigs.k8s.io/kubefed v0.1.0-rc6.0.20200224204536-6207193c49f7
+	sigs.k8s.io/yaml v1.2.0 // indirect
 )
 
+replace github.com/codeready-toolchain/api => github.com/alexeykazakov/api v0.0.0-20200306233032-6a4378dbcd57
```

Comment:
Will need to remove it when https://github.com/codeready-toolchain/api/pull/118 is merged.

Reply:
> @xcoulon the test were failing because the team tier introduced by Matous didn't have cluster resources so the generate failed to create that tier. I made a few changes:
> 
> * made cluster resource truly optional
> * added some tests for that ^
> * added cluster resources to the team tier. I think we should support tiers without cluster resources but in this particular case we do need to specify cluster resources for that tier.
> 
> Please review my changes!

I don't understand why we can't simply change the tests?

Also, maybe we could change the `ClusterResources NSTemplateTierClusterResources` field in the `NSTemplateTierSpec` type to `ClusterResources []NSTemplateTierClusterResources` (ie, an array) which means:
- we wouldn't have to use a pointer (I'm a bit reluctant to that, but that's my personal opinion) 
- we could have something more structured with subfolders as we discussed in https://github.com/codeready-toolchain/host-operator/pull/157/files#r388772370, with the possibility of multiple files in the `cluster` subdir in each "tier" folder.

WDYT @alexeykazakov and @MatousJobanek ?